cypress/e2e/admin/project_settings.cy.ts

```diff
@@ -0,0 +1,804 @@
+        .should("have.value", min)
+        .get_by_testid(ElementId.max)
+        .should("have.value", max);
+
+      if (disallow) {
+        cy.get_by_testid(ElementId.disallow).should("be.checked");
+      } else {
+        cy.get_by_testid(ElementId.disallow).should("not.be.checked");
+      }
+    };
+
+    cy.visit(this.page_uri);
+    assert_correct_values(1, 1, false);
+
+    cy.get_by_testid(ElementId.max)
+      .should("be.visible")
+      .type("{moveToEnd}{backspace}12");
+    assert_correct_values(1, 12, false);
+    cy.save_and_reload();
+    assert_correct_values(1, 12, false);
+
+    cy.get_by_testid(ElementId.min)
+      .should("be.visible")
+      .type("{moveToEnd}{backspace}2");
+    assert_correct_values(2, 12, false);
+    cy.save_and_reload();
+    assert_correct_values(2, 12, false);
+
+    cy.get_by_testid(ElementId.disallow).should("be.visible").check();
+    assert_correct_values(2, 12, true);
+    cy.save_and_reload();
+    assert_correct_values(2, 12, true);
+
+    cy.get_by_testid(ElementId.max)
+      .should("be.visible")
+      .type("{moveToEnd}{backspace}{backspace}9");
+    assert_correct_values(2, 9, true);
+    cy.save_and_reload();
+    assert_correct_values(2, 9, true);
+
+    cy.get_by_testid(ElementId.min)
+      .should("be.visible")
+      .type("{moveToEnd}{backspace}3");
+    assert_correct_values(3, 9, true);
+    cy.save_and_reload();
+    assert_correct_values(3, 9, true);
+
+    cy.get_by_testid(ElementId.disallow).should("be.visible").uncheck();
+    assert_correct_values(3, 9, false);
+    cy.save_and_reload();
+    assert_correct_values(3, 9, false);
+  });
+
+  it("shows group settings related API errors", function () {
+    cy.visit(this.page_uri)
+      .get_by_testid("min-group-size")
+      .type("{backspace}22")
+      .save()
+      .get_api_errors()
+      .should("have.length", 1)
+      .should("contain.text", "group size");
+
+    // make sure error goes away after valid request
+    cy.get_by_testid("max-group-size").type("{backspace}33").save_and_reload();
+  });
+
+  it("does not allow the user to update group settings with invalid values", function () {
+    enum ElementId {
+      min = "min-group-size",
+      min_err = "min-group-size-error",
+      max = "max-group-size",
+      max_err = "max-group-size-error",
+      disallow = "disallow-group-registration",
+      save = "save-button",
+    }
+    cy.visit(this.page_uri);
+
+    cy.get_by_testid(ElementId.min)
+      .should("be.visible")
+      .type("10a")
+      .get_by_testid(ElementId.min_err)
+      .should("contain.text", "integer")
+      .get_by_testid(ElementId.save)
+      .should("be.disabled")
+      .click({ force: true })
+      .get_by_testid(ElementId.max)
+      .should("be.visible")
+      .type("b0")
+      .get_by_testid(ElementId.max_err)
+      .should("contain.text", "integer")
+      .get_by_testid(ElementId.save)
+      .should("be.disabled")
+      .click({ force: true })
```

Comment:
Is this basically saying "act as if the user clicked the mouse on the disabled button", with the expectation that nothing happens (i.e., the form is not submitted)?

Reply:
Yes. Maybe a silly thing to test 🤷